tests/core_tests/chaingen.h

```diff
@@ -426,13 +426,16 @@ bool construct_miner_tx_manually(size_t height, uint64_t already_generated_coins
                                  uint64_t fee, uint8_t hf_version = 1,
                                  cryptonote::keypair* p_txkey = nullptr);
 
+extern bool tx_construction_check_unlock_time;
```

Comment:
This is not ideal. It is a global default value for `bool check_unlock_time` parameter to `construct_tx_to_key` functions below. By default, it is set to false so backward compatibility with all core_tests is not affected. (Setting it to true causes few tests to fail).

But for Trezor tests I need transaction to pass full daemon validation, i.e., there cannot be fake outputs which are still locked. This global variable enables Trezor tests to globally enable unlock time checking. Function caller can still provide own `check_unlock_time` value.

Another approach would be to modify all TX creation macros, create extended versions which would enable the unlock time checking. It would increase code size significantly.

Reply:
Change: I fixed all related tests instead, resolving